src/storage/temporary_file_manager.cpp

```diff
@@ -224,8 +252,23 @@ bool TemporaryFileIndex::IsValid() const {
 // TemporaryFileManager
 //===--------------------------------------------------------------------===//
 
+static idx_t GetDefaultMax(const string &path) {
+	// Use the available disk space
+	auto disk_space = FileSystem::GetAvailableDiskSpace(path);
+	idx_t default_value = 0;
+	if (disk_space.IsValid()) {
+		// Only use 90% of the available disk space
+		default_value = static_cast<idx_t>(static_cast<double>(disk_space.GetIndex()) * 0.9);
+	}
+	return default_value;
```

Comment:
Hmm fair, it might be better to make an assertion out of this though
Since we are ensuring that the file exists here:
```c++
	if (!temp_directory.empty()) {
		if (!fs.DirectoryExists(temp_directory)) {
			fs.CreateDirectory(temp_directory);
			created_directory = true;
		}
	}
	temp_file->SetMaxSwapSpace(max_swap_space);
```

Reply:
This code feels needlessly complex, why are we making a TemporaryDirectoryHandle when no path is specified?